google/cloud/aiplatform/matching_engine/matching_engine_index.py

```diff
@@ -0,0 +1,596 @@
+                contain lowercase letters, numeric characters,
+                underscores and dashes. International characters
+                are allowed.
+                See https://goo.gl/xmQnxf for more information
+                on and examples of labels. No more than 64 user
+                labels can be associated with one
+                Index(System labels are excluded)."
+                System reserved label keys are prefixed with
+                "aiplatform.googleapis.com/" and are immutable.
+            project (str):
+                Optional. Project to create EntityType in. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to create EntityType in. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to create EntityTypes. Overrides
+                credentials set in aiplatform.init.
+            request_metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as metadata.
+            encryption_spec (str):
+                Optional. Customer-managed encryption key
+                spec for data storage. If set, both of the
+                online and offline data storage will be secured
+                by this key.
+            sync (bool):
+                Optional. Whether to execute this creation synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
+
+        Returns:
+            MatchingEngineIndex - Index resource object
+
+        """
+
+        algorithm_config = matching_engine_index_config.TreeAhConfig(
+            leaf_node_embedding_count=leaf_node_embedding_count,
+            leaf_nodes_to_search_percent=leaf_nodes_to_search_percent,
+        )
+
+        config = matching_engine_index_config.MatchingEngineIndexConfig(
+            dimensions=dimensions,
+            algorithm_config=algorithm_config,
+            approximate_neighbors_count=approximate_neighbors_count,
+            distance_measure_type=distance_measure_type,
+        )
+
+        return cls._create(
+            display_name=display_name,
+            contents_delta_uri=contents_delta_uri,
+            config=config,
+            description=description,
+            labels=labels,
+            project=project,
+            location=location,
+            credentials=credentials,
+            request_metadata=request_metadata,
+            sync=sync,
+        )
+
+    @classmethod
+    def create_brute_force_index(
+        cls,
+        display_name: str,
+        contents_delta_uri: str,
+        dimensions: int,
+        approximate_neighbors_count: int,
```

Comment:
Can we change this to optional? This field is only required for TreeAh.

Reply:
We can remove if not needed by `brute_force`

Reply:
Removed.